src/Sentry/Protocol/Envelopes/EnvelopeItem.cs

```diff
@@ -128,77 +128,49 @@ await using (writer.ConfigureAwait(false))
             writer.Flush();
         }
 
-        private Task SerializeHeaderAsync(
-            Stream stream,
-            IDiagnosticLogger? logger,
-            CancellationToken cancellationToken) =>
-            SerializeHeaderAsync(stream, Header, logger, cancellationToken);
-
-        private void SerializeHeader(Stream stream, IDiagnosticLogger? logger) =>
-            SerializeHeader(stream, Header, logger);
-
         /// <inheritdoc />
-        public async Task SerializeAsync(Stream stream, IDiagnosticLogger? logger, CancellationToken cancellationToken = default)
+        public async Task SerializeAsync(Stream stream, IDiagnosticLogger? logger,
+            CancellationToken cancellationToken = default)
         {
-            // Length is known
-            if (TryGetLength() != null)
-            {
-                // Header
-                await SerializeHeaderAsync(stream, logger, cancellationToken).ConfigureAwait(false);
-                await stream.WriteByteAsync((byte)'\n', cancellationToken).ConfigureAwait(false);
+            // Always calculate the length of the payload, as Sentry will reject envelopes that have incorrect lengths
```

Comment:
If debug mode is enabled should we assert the values we recalculated and the original match and otherwise log? It can help us find discrepancies in round trip serialization

Reply:
No, because it's *valid* for there to be such discrepancies.  That's what this issue is about, essentially.